mongomock/store.py

```diff
@@ -78,29 +82,97 @@ def is_created(self):
     def drop(self):
         self._documents = collections.OrderedDict()
         self.indexes = {}
+        self._ttl_indexes = {}
         self._is_force_created = False
 
+    def create_index(self, index_name, index_dict):
+        self.indexes[index_name] = index_dict
+        if index_dict.get('expireAfterSeconds') is not None:
+            self._ttl_indexes[index_name] = index_dict
+
+    def drop_index(self, index_name):
+        # The main index object should raise a KeyError, but the
+        # TTL indexes have no meaning to the outside.
+        del self.indexes[index_name]
+        self._ttl_indexes.pop(index_name, None)
+
     @property
     def is_empty(self):
+        self._remove_expired_documents()
         return not self._documents
 
     def __contains__(self, key):
+        self._remove_expired_documents()
         return key in self._documents
 
     def __getitem__(self, key):
+        self._remove_expired_documents()
         return self._documents[key]
 
     def __setitem__(self, key, val):
         with lock:
             self._documents[key] = val
 
     def __delitem__(self, key):
-        del self._documents[key]
+        self._documents.pop(key)
```

Comment:
why the change here? I find the `del` version more readable.

Also, shouldn't you need to remove expired documents before removing the key? I guess we are missing tests around deleting non existing documents.